### PR TITLE
[Cache] Add TLS scheme for Redis connection

### PR DESCRIPTION
**Pull request by misaert**
See https://github.com/symfony/symfony/pull/39599.

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Is that correct? Isn't TLS enabled by using the `rediss` scheme instead?

**Comment by misaert-goto-games** [comment]
No, it is not correct because the implementation has changed over time. I take care of changing this PR to correspond with the choosen implementation :wink:

**Comment by misaert** [comment]
No, it is not correct because the implementation has changed over time. I take care of changing this PR to correspond with the choosen implementation :wink:

**Comment by misaert** [comment]
Changed. Do I have to change the destination of the branch to `4.4` one?

**Comment by nicolas-grekas** [comment]
Yes please, and rebase on 4.4 before.

**Comment by misaert** [comment]
Done.

File: components/cache/adapters/redis_adapter.rst [hunk 1]
```diff
@@ -62,15 +62,16 @@ helper method allows creating and configuring the Redis client class instance us
     );
 
 The DSN can specify either an IP/host (and an optional port) or a socket path, as well as a
-password and a database index.
+password and a database index. To enable the transport level security for connections, the scheme `redis` must be
```
Comment (OskarStark) [inline components/cache/adapters/redis_adapter.rst]: Please 2 backticks, same on the next line, thanks

Comment (misaert-goto-games) [reply components/cache/adapters/redis_adapter.rst]: Done :+1:

**Review by OskarStark — APPROVED** [review]

File: components/cache/adapters/redis_adapter.rst (hunk 1 above)
Comment (misaert) [reply components/cache/adapters/redis_adapter.rst]: Done :+1:

**Review by maxhelias — APPROVED** [review]

File: components/cache/adapters/redis_adapter.rst [hunk 2]
```diff
@@ -62,15 +62,16 @@ helper method allows creating and configuring the Redis client class instance us
     );
 
 The DSN can specify either an IP/host (and an optional port) or a socket path, as well as a
-password and a database index.
+password and a database index. To enable the transport level security for connections, the scheme ``redis`` must be
```
Comment (nicolas-grekas) [inline components/cache/adapters/redis_adapter.rst]: To enable TLS for connections [...]

Comment (misaert) [reply components/cache/adapters/redis_adapter.rst]: Done.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by wouterj** [comment]
Hi @misaert! I'm not sure why this PR has been open for quite a while after the code merge. All seems perfect. Thanks for taking care of updating the docs!

**Pull request merged**